test/darray.jl

```diff
@@ -839,6 +839,16 @@ check_leaks()
     f = 2 .* e
     @test Array(f) == 2 .* Array(e)
     @test Array(map(x -> sum(x) .+ 2, e)) == map(x -> sum(x) .+ 2, e)
+
+    @testset "test nested broadcast" begin
+       g = a .- m .* sin.(c)
+       @test Array(g) == Array(a) .- Array(m) .* sin.(Array(c))
+    end
+
+    # @testset "lazy wrapped broadcast" begin
```

Comment:
Why is this commented out?

Reply:
Test fail with:
```
 MethodError: no method matching localpartindex(::SubArray{Float64,2,DArray{Float64,2,Array{Float64,2}},Tuple{UnitRange{Int64},Base.Slice{Base.OneTo{Int64}}},false})
  Closest candidates are:
    localpartindex(!Matched::Array{Int64,N} where N) at /home/vchuravy/.julia/dev/DistributedArrays/src/darray.jl:306
    localpartindex(!Matched::DArray) at /home/vchuravy/.julia/dev/DistributedArrays/src/darray.jl:314
  #117 at /home/vchuravy/.julia/dev/DistributedArrays/src/broadcast.jl:65
  #112 at /build/julia/src/julia/usr/share/julia/stdlib/v1.0/Distributed/src/process_messages.jl:269
  run_work_thunk at /build/julia/src/julia/usr/share/julia/stdlib/v1.0/Distributed/src/process_messages.jl:56
  macro expansion at /build/julia/src/julia/usr/share/julia/stdlib/v1.0/Distributed/src/process_messages.jl:269 [inlined]
  #111 at ./task.jl:259
```

and I am not sure yet how to do `localpart(view(Darray, ...))`